### PR TITLE
Cluster agent built for FIPS

### PR DESCRIPTION
**Pull request by ktsakalozos**
Same as k8s-dqlite and k8s services. Building the cluster-agent with cgo support and having an env file to load the right libs before starting the service.

### CODE REVIEW

File: microk8s-resources/default-args/cluster-agent-env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# In a node where FIPS is enabled, ie /proc/sys/crypto/fips_enabled is set to 1
+# Force FIPS to be used with. If not set GOFIPS is implicetely selected.
+export GOFIPS=0
```
Comment (neoaggelos) [inline microk8s-resources/default-args/cluster-agent-env:3]: Should this be commented as well?

**Review by neoaggelos — CHANGES_REQUESTED** [review]
Can we have the non-fips parts into master instead? Namely, the `cluster-agent-env` file.

File: build-scripts/components/cluster-agent/build.sh [hunk 2]
```diff
@@ -4,5 +4,6 @@ export INSTALL="${1}/bin"
 export GOEXPERIMENT=opensslcrypto
 mkdir -p "${INSTALL}"
 
-make cluster-agent
+go mod tidy
```
[the remark this answers is not in the document]
Comment (ktsakalozos) [reply build-scripts/components/cluster-agent/build.sh]: Just blindly copied the Makefile. I removed it.

File: microk8s-resources/default-args/cluster-agent-env (hunk 1 above)
Comment (ktsakalozos) [reply microk8s-resources/default-args/cluster-agent-env:3]: No. The idea here is that even in a fips capable system we need to start normally (ignore FIPS) and then if the user needs to use FIPS she would need to point MicroK8s to the right libs.

File: microk8s-resources/wrappers/run-cluster-agent-with-args [hunk 3]
```diff
@@ -14,6 +14,13 @@ export LD_LIBRARY_PATH="$SNAP/lib:$SNAP/usr/lib:$SNAP/lib/$ARCH-linux-gnu:$SNAP/
 export LD_LIBRARY_PATH=$SNAP_LIBRARY_PATH:$LD_LIBRARY_PATH
 export OPENSSL_CONF="$SNAP/etc/ssl/openssl.cnf"
 
+set -a
+if [ -e "${SNAP_DATA}/args/cluster-agent-env" ]
+then
+  . "${SNAP_DATA}/args/cluster-agent-env"
+fi
+set +a
+
```
Comment (neoaggelos) [inline microk8s-resources/wrappers/run-cluster-agent-with-args:23]: We should have this on `master` as well, not only `feature/fips`

**Review by neoaggelos — APPROVED** [review]
LGTM, repeating my one comment for cluster-agent-env

**Pull request merged**